src/Gzero/Core/Parsers/ArrayParser.php

```diff
@@ -0,0 +1,156 @@
+<?php namespace Gzero\Core\Parsers;
+
+use Gzero\Core\Query\QueryBuilder;
+use Gzero\InvalidArgumentException;
+use Illuminate\Http\Request;
+
+class ArrayParser implements ConditionParser {
+
+    /** @var string */
+    protected $name;
+
+    /** @var string */
+    protected $operation = 'in';
+
+    /** @var mixed */
+    protected $value;
+
+    /** @var bool */
+    protected $applied = false;
+
+    /** @var array */
+    protected $availableOperations = ['in', 'not in'];
+
+    /** @var array */
+    protected $option;
+
+    /**
+     * @param string $name    Field name
+     *
+     * @param array  $options Optional array of options
+     *
+     * @throws InvalidArgumentException
+     */
+    public function __construct(string $name, $options = [])
+    {
+        if (empty($name)) {
+            throw new InvalidArgumentException('ArrayParser: Name must be defined');
+        }
+        $this->name   = $name;
+        $this->option = $options;
+    }
+
+    /**
+     * It returns field name
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * It returns operation
+     *
+     * @return string
+     */
+    public function getOperation(): string
+    {
+        return $this->operation;
+    }
+
+    /**
+     * It returns value
+     *
+     * @return mixed|null
+     */
+    public function getValue()
+    {
+        return ($this->value) ?: null;
+    }
+
+    /**
+     * Checks if field was present in response during parse phase
+     *
+     * @return bool
+     */
+    public function wasApplied(): bool
+    {
+        return $this->applied;
+    }
+
+    /**
+     * It parses request field
+     *
+     * @param Request $request Request object
+     *
+     * @return void
+     *
+     * @throws InvalidArgumentException
+     */
+    public function parse(Request $request)
+    {
+        if ($request->has($this->name)) {
+            $this->applied = true;
+            $value         = $request->input($this->name);
+
+            if (substr($value, 0, 1) === '!') {
+                $this->operation = 'not in';
+
+                $this->value = explode(',', substr($value, 1));
+            } elseif (empty($value)) {
+                $this->value = null;
+            } else {
+                $this->value = explode(',', $value);
+            }
+
+            // Need it because of Cyclomatic Complexity.
```

Comment:
Cyclomatic complexity?
I don't quite understand what you mean by it here?

Reply:
@bartero Level of cyclomatic complexity is checked by phpmd and it's set to 8, so in order to avoid this error being reported for the `parse` method, part of the logic is extracted to checkValue.

Reply:
That's very strange. I will check-run it later.

Reply:
@michalstrzelecki @bartero Maybe it's not the case here - it's just a copy/paste issue from more complex parser?

Reply:
If I paste `postProcessing` method's body in parse method I have got cyclomatic complexity warnings.